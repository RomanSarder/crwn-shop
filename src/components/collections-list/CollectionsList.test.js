import React from 'react'
import { render } from '@testing-library/react'
import configureStore from 'redux-mock-store'
import CollectionsListContainer, { CollectionsList } from './CollectionsList'
import { Provider } from 'react-redux'

jest.mock('../collection/Collection', () => () => (<div>Hello World</div>))

var testCollections = [
    {
        id: 1,
        name: 'test 1',
        items: []
    },
    {
        id: 2,
        name: 'test 2',
        items: []
    }
]

function renderCollectionsList () {
    var renderResult = render(
        <CollectionsList collections={testCollections} />
    )

    return renderResult
}

function renderCollectionsListContainer (store) {
    var renderResult = render(
        <Provider store={store}>
            <CollectionsListContainer />
        </Provider>
    )

    return renderResult
}

it('<CollectionsList /> should render with mocked collections', () => {
    var { container, getAllByText } = renderCollectionsList()

    expect(container.firstChild).toMatchSnapshot()
    expect(getAllByText('Hello World').length).toEqual(2)
})

it('<CollectionsList />  should render with redux state', () => {
    var store = configureStore()({
        collections: {
            data: {
                [testCollections[0].name]: {
                    id: 1,
                    items: []
                },
                [testCollections[1].name]: {
                    id: 2,
                    items: []
                }
            }
        }
    })
    var { getAllByText, container } = renderCollectionsListContainer(store)
    expect(container.firstChild).toMatchSnapshot()
    expect(getAllByText('Hello World').length).toEqual(2)
})