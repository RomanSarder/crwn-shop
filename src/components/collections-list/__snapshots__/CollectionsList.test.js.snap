// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CollectionsList />  should render with redux state 1`] = `
<div
  class="sc-bdvvaa havjpZ"
>
  <div
    class="collections"
  >
    <div>
      Hello World
    </div>
    <div>
      Hello World
    </div>
  </div>
</div>
`;

exports[`<CollectionsList /> should render with mocked collections 1`] = `
<div
  class="sc-bdvvaa havjpZ"
>
  <div
    class="collections"
  >
    <div>
      Hello World
    </div>
    <div>
      Hello World
    </div>
  </div>
</div>
`;
